server/rehype-hljs-var.ts

```diff
@@ -35,46 +24,47 @@ export const rehypeVarInHLJS = (
   return (root: Parent, file: VFile) => {
     const highlighter = rehypeHighlight(options);
 
-    let placeholdersToVars: Record<string, MdxJsxFlowElement> = {};
+    let placeholdersToVars: Record<string, Node> = {};
 
     // In a code snippet, Var elements are parsed as text. Replace these with
     // UUID strings to ensure that the parser won't split these up and make
     // them unrecoverable.
     visit(root, undefined, (node: Node, index: number, parent: Parent) => {
+      // We only visit text nodes inside code snippets
       if (
-        node.type === "text" &&
-        parent.hasOwnProperty("tagName") &&
-        (parent as Element).tagName === "code"
+        node.type !== "text" ||
+        !parent.hasOwnProperty("tagName") ||
+        (parent as Element).tagName !== "code"
       ) {
-        const varPattern = new RegExp("<Var [^>]+/>", "g");
-        (node as Text).value = (node as Text).value.replace(
-          varPattern,
-          (match) => {
-            const placeholder = makePlaceholder();
-            // Since the Var element was originally text, parse it so we can recover
-            // its properties. The result should be a small HTML AST with a root
-            // node and one child, the Var node.
-            const varElement = unified()
-              .use(remarkParse)
-              .use(remarkMDX)
-              .parse(match);
-            if (
-              varElement.children.length !== 1 ||
-              (varElement.children[0] as MdxJsxFlowElement).name !== "Var"
-            ) {
-              throw new Error(
-                `Problem parsing file ${file.path}: malformed Var element within a code snippet`
-              );
-            }
-
-            placeholdersToVars[placeholder] = varElement
-              .children[0] as MdxJsxFlowElement;
-            return placeholder;
-          }
-        );
+        return [CONTINUE];
```

Comment:
what a weird pattern to return a const like that in an array lol, never seen anyhting like it.

Reply:
It's a tuple to control AST traversal: https://www.npmjs.com/package/unist-util-visit#visittree-test-visitor-reverse

Yeah, pretty odd!